src/item.cpp

```diff
@@ -297,6 +262,134 @@ std::string Item::PrettyName() const {
+                category_vector_.push_back("OneHand");
+                category_ = "Axe";
+            } else if(category_ == "Onemace") {
+                category_vector_.push_back("OneHand");
+                category_ = "Mace";
+            } else if(category_ == "Onesword") {
+                category_vector_.push_back("OneHand");
+                category_ = "Sword";
+            } else if(category_ == "Twoaxe") {
+                category_vector_.push_back("TwoHand");
+                category_ = "Axe";
+            } else if(category_ == "Twomace") {
+                category_vector_.push_back("TwoHand");
+                category_ = "Mace";
+            } else if(category_ == "Twosword") {
+                category_vector_.push_back("TwoHand");
+                category_ = "Sword";
+            }
+
+            category_vector_.push_back(category_);
+        } else if(category_ == "Maps") {
+            // Use icon path to determine possible expansion sub-category
+            std::smatch sm;
+            if(std::regex_search(icon_, sm, std::regex("/Art/2DItems/Maps/(.*)/"))) {
+                std::string match = sm.str(1);
+                std::vector<std::string> iconsubs;
+                boost::split(iconsubs, match, boost::is_any_of("/"));
+                std::string sub = iconsubs[0];
+                if(sub == "Atlas2Maps") {
+                    category_vector_.push_back("3.1");
+                    if(iconsubs.size() > 1) {
+                        sub = iconsubs[1];  // Typically "New"
+                        Util::Capitalise(sub);
+                        category_vector_.push_back(sub);
+                    }
+                } else if(sub == "AtlasMaps") {
+                    category_vector_.push_back("2.4");
+                } else if(sub == "act4maps") {
+                    category_vector_.push_back("2.0");
+                } else {
+                    Util::Capitalise(sub);
+                    category_vector_.push_back(sub);
+                }
+            } else {
+                if(icon_.find("/Art/2DItems/Maps/Map") != std::string::npos) {
+                    // Doesn't find all because of some maps like FairgravesMap01.png, olmec.png, etc. They'll end up in Maps.Misc, then moved to Maps.Older Uniques
+                    category_vector_.push_back("Original");
+                } else if(icon_.find("/Art/2DItems/Currency/Breach/") != std::string::npos) {
+                    // Isn't really a map, has no property named Map Tier
+                    category_vector_.pop_back();
+                    category_vector_.push_back("Currency");
+                    category_vector_.push_back("Breach");
+                } else {
+                    category_vector_.push_back("Misc");
+
+                    if(json.HasMember("properties") && json["properties"].IsArray()) {
+                        for(auto prop_it = json["properties"].Begin(); prop_it != json["properties"].End(); ++prop_it) {
```

Comment:
Use a range for instead, `for (auto &prop : json["properties"])`

Reply:
IIRC this was a copy-paste from the existing code that handles the properties array in general, so I'd change both if it's now not acceptable.